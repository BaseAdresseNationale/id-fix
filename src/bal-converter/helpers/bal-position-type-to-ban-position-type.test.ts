import { describe, expect, test } from "vitest";
import { convertBalPositionTypeToBanPositionType } from './index' 

describe("balPositionTypeToBanPositionType", () => {

  test("should return undefined", async () => {
    expect(convertBalPositionTypeToBanPositionType('')).toMatchSnapshot();
  });

  test("should return undefined", async () => {
    expect(convertBalPositionTypeToBanPositionType('not-in-dictionary')).toMatchSnapshot();
  });

  test("should return entrance", async () => {
    expect(convertBalPositionTypeToBanPositionType('entrée')).toMatchSnapshot();
  });

  test("should return building", async () => {
    expect(convertBalPositionTypeToBanPositionType('bâtiment')).toMatchSnapshot();
  });

  test("should return staircase identifier", async () => {
    expect(convertBalPositionTypeToBanPositionType('cage d’escalier')).toMatchSnapshot();
  });

  test("should return utility service", async () => {
    expect(convertBalPositionTypeToBanPositionType('service technique')).toMatchSnapshot();
  });

  test("should return postal delivery", async () => {
    expect(convertBalPositionTypeToBanPositionType('délivrance postale')).toMatchSnapshot();
  });

  test("should return parcel", async () => {
    expect(convertBalPositionTypeToBanPositionType('parcelle')).toMatchSnapshot();
  });

  test("should return segment", async () => {
    expect(convertBalPositionTypeToBanPositionType('segment')).toMatchSnapshot();
  });

  test("should return other", async () => {
    expect(convertBalPositionTypeToBanPositionType('autre')).toMatchSnapshot();
  });
})