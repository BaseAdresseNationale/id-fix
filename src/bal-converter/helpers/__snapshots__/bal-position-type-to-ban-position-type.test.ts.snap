// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`balPositionTypeToBanPositionType > should return building 1`] = `"building"`;

exports[`balPositionTypeToBanPositionType > should return entrance 1`] = `"entrance"`;

exports[`balPositionTypeToBanPositionType > should return other 1`] = `"other"`;

exports[`balPositionTypeToBanPositionType > should return parcel 1`] = `"parcel"`;

exports[`balPositionTypeToBanPositionType > should return postal delivery 1`] = `"postal delivery"`;

exports[`balPositionTypeToBanPositionType > should return segment 1`] = `"segment"`;

exports[`balPositionTypeToBanPositionType > should return staircase identifier 1`] = `"staircase identifier"`;

exports[`balPositionTypeToBanPositionType > should return undefined 1`] = `undefined`;

exports[`balPositionTypeToBanPositionType > should return undefined 2`] = `undefined`;

exports[`balPositionTypeToBanPositionType > should return utility service 1`] = `"utility service"`;
